soc/arm/gigadevice/gd32a50x/soc.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2022 YuLong Yao <feilongphone@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _SOC_ARM_GIGADEVICE_GD32A5XX_SOC_H_
+#define _SOC_ARM_GIGADEVICE_GD32A5XX_SOC_H_
+
+#ifndef _ASMLANGUAGE
+
+#include <gd32a50x.h>
+
+/* The GigaDevice HAL headers define this, but it conflicts with the Zephyr can.h */
+#undef CAN_MODE_NORMAL
```

Comment:
Please follow [Current known conflict list](https://github.com/zephyrproject-rtos/hal_gigadevice#current-known-conflict-list).

Reply:
Looks this just a outdated code, remove it.